src/gf.c

```diff
@@ -2464,7 +2476,7 @@ static int ml_matches_visitor(jl_typemap_entry_t *ml, struct typemap_intersectio
         // NOTE: jl_subtype check added in case the intersection is
         // over-approximated.
         if (matched_all_typevars && jl_types_equal(closure->match.ti, closure->match.type) &&
-            jl_subtype(closure->match.type, (jl_value_t*)ml->sig, 0)) {
+            jl_subtype(closure->match.type, (jl_value_t*)ml->sig)) {
```

Comment:
in the comment above this, I think the `implies` arrow is reversed (which would agree with the note just below it). maybe this should just be the subtype test?